src/internal/carriers/pickups/pickup-package.ts

```diff
@@ -38,8 +39,15 @@ export class PickupPackage extends PackageIdentifierBase implements IPickupPacka
     try {
       pkg = app[_internal].references.lookup(pojo.packaging, Packaging);
     } catch {
-      if (typeof pojo.packaging === 'string') {
-        pkg = pojo.packaging;
+      if (typeof pojo.packaging === "string") {
+        pkg = new Packaging(
+          {
+            id: v4(),
+            name: pojo.packaging,
+            description: pojo.packaging,
+            code: "custom"
+          }
+        );
       }
     }
```

Comment:
This same block of code (the entire try...catch block) is repeated in 4 places.  Move it to a utility function instead.

Reply:
done.